src/app.ts

```diff
@@ -63,13 +65,17 @@ export const handler = async (event: CloudFrontRequestEvent): Promise<CloudFront
       logger,
       suffix,
     })
-  } else if (pathname === (await getStatusUri(customerVariables))) {
-    return handleStatus(customerVariables)
   } else {
-    return new Promise((resolve) =>
-      resolve({
-        status: '404',
-      }),
-    )
+    const statusUri = await getStatusUri(customerVariables)
+
+    if (pathname === statusUri) {
+      return handleStatus(customerVariables)
+    } else {
```

Comment:
I think it improves the readability if we remove `else` here, and just put `return` at bottom.

Reply:
Good catch, done!